marklogic-data-hub-central/ui/src/components/modeling/graph-view/properties-tab/properties-tab.tsx

```diff
@@ -1,22 +1,61 @@
-import React from "react";
+import React, {useState} from "react";
 import PropertyTable from "../../property-table/property-table";
+import styles from "./properties-tab.module.scss";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faLayerGroup, faKey} from "@fortawesome/free-solid-svg-icons";
+import arrayIcon from "../../../../assets/icon_array.png";
 
 interface Props {
   entityTypeData: any;
   canWriteEntityModel: any;
   canReadEntityModel: any;
 }
 
+
 const PropertiesTab: React.FC<Props> = (props) => {
+  const [showLegend, setShowLegend] = useState(false);
+
+  const toggleLegend = () => {
+    if (showLegend) {
+      setShowLegend(false);
+    } else {
+      setShowLegend(true);
+    }
+  };
+
+  const LegendShowHide =
+    <div>
+      {showLegend ?
+        <div className={styles.legend}>
+          <div data-testid="foreignKeyIconLegend" className={styles.foreignKeyLegendText}>
+            <FontAwesomeIcon className={styles.foreignKeyIcon} icon={faKey}/> Foreign
+                                  Key Relationship
+          </div>
+          <div data-testid="multipleIconLegend" className={styles.legendText}><img
+            className={styles.arrayImage} src={arrayIcon} alt={""}/> Multiple
+          </div>
+          <div data-testid="structuredIconLegend" className={styles.legendText}>
+            <FontAwesomeIcon className={styles.structuredIcon}
+              icon={faLayerGroup}/> Structured Type
+          </div>
+          <a data-testid="hideLegendLink" onClick={() => toggleLegend()}>&nbsp;&nbsp;&nbsp;Hide Legend &lt;&lt;</a>
```

Comment:
Using &nbsp; for layout (e.g. padding) is generally an anti-pattern and in this case results in clickable (linked) whitespace, which seems odd. Can we remove those HTML entities and format using CSS where needed?

Reply:
Right thanks, removed the &nbsp; as it's actually redundant. I believe the &lt; and &gt; are necessary though to display "<<" and ">>" without react picking it up as a tag.

Reply:
Yeah, I just meant the space entities, thanks!